cmd/eval.go

```diff
@@ -17,80 +17,143 @@ limitations under the License.
 package cmd
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/hacbs-contract/ec-cli/cmd/internal/applicationsnapshot"
 	"github.com/hacbs-contract/ec-cli/cmd/internal/image"
 	"github.com/hacbs-contract/ec-cli/cmd/internal/policy"
+	appstudioshared "github.com/redhat-appstudio/managed-gitops/appstudio-shared/apis/appstudio.redhat.com/v1alpha1"
 	"github.com/spf13/cobra"
 )
 
-var arguments = struct {
-	policyConfiguration string
-	imageRef            string
-	publicKey           string
-	rekorURL            string
-	strict              bool
-}{
-	rekorURL: "https://rekor.sigstore.dev/",
+func init() {
+	rootCmd.AddCommand(evalCmd())
 }
 
-func init() {
-	rootCmd.AddCommand(evalCmd)
+func evalCmd() *cobra.Command {
+	var arguments = struct {
+		policyConfiguration string
+		imageRef            string
+		publicKey           string
+		rekorURL            string
+		strict              bool
+		input               string
+		filepath            string
+		output              string
+	}{
+		rekorURL: "https://rekor.sigstore.dev/",
+	}
+	var snapshotSpec *appstudioshared.ApplicationSnapshotSpec
+
+	evalCmd := &cobra.Command{
+		Use:   "eval",
+		Short: "Evaluate enterprise contract",
+		Long:  `TODO: description`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if arguments.input != "" || arguments.filepath != "" {
+				var jsonstr string
+				// Filepath handler, reads the file and stores the content as string
+				if arguments.filepath != "" {
+					file, err := ioutil.ReadFile(arguments.filepath)
+
+					if err != nil {
+						return err
+					}
+
+					jsonstr = string(file)
+				}
+
+				// Input handler
+				if arguments.input != "" {
+					jsonstr = arguments.input
+				}
+				// Unmarshall json into struct, exit on failure
+				if err := json.Unmarshal([]byte(jsonstr), &snapshotSpec); err != nil {
```

Comment:
Hmm, is this `**appstudioshared.ApplicationSnapshotSpec`? Looks odd given _If v is nil or not a pointer, Unmarshal returns an InvalidUnmarshalError._ from the docs of [json.Unmarshal](https://pkg.go.dev/encoding/json#Unmarshal).
I think we're missing an unit test for this to make sure.